kubernetes/examples/FasterTransformer_HuggingFace_Bert.md

```diff
@@ -0,0 +1,279 @@
+```
+
+7. Generate model files
+
+```bash
+cd examples/Huggingface_Transformers
+python Download_Transformer_models.py
+
+cd /workspace/FasterTransformer/build/
+
+python pytorch/Bert_FT_trace.py --mode sequence_classification --model_name_or_path "/workspace/serve/examples/Huggingface_Transformers/Transformer_model" --tokenizer_name "bert-base-uncased" --batch_size 1 --data_type fp16 --model_type thsext
+
+cd /workspace/serve/examples/Huggingface_Transformers
+
+# make sure to change the ../Huggingface_Transformers/setup_config.json "save_mode":"torchscript" and "FasterTransformer":true
+
+# change the ../Huggingface_Transformers/setup_config.json
+
+{
+"model_name":"bert-base-uncased",
+"mode":"sequence_classification",
+"do_lower_case":true,
+"num_labels":"0",
+"save_mode":"pretrained",
+"max_length":"128",
+"captum_explanation":false,
+"embedding_name": "bert",
+"FasterTransformer":true
+}
+
+# add a requirement.txt with transformers==2.5.1
+
+echo "transformers==2.5.1" >> requirements.txt
+```
+
+8. Create Mar file
+
+```bash
+torch-model-archiver --model-name BERTSeqClassification --version 1.0 --serialized-file /workspace/FasterTransformer/build/traced_model.pt --handler ../Huggingface_Transformers/Transformer_handler_generalized.py --extra-files "../Huggingface_Transformers/setup_config.json,../Huggingface_Transformers/Seq_classification_artifacts/index_to_name.json,/workspace/FasterTransformer/build/lib/libpyt_fastertransformer.so" --requirements-file requirements.txt
+```
+
+9. Exit container
+
+## Copy Mar file from container to local path
+
+```bash
+docker cp <container-id>:/workspace/serve/examples/Huggingface_Transformers/BERTSeqClassification.mar ./BERTSeqClassification.mar
+```
+
+## Create config.properties
+
+```bash
+inference_address=http://0.0.0.0:8080
+management_address=http://0.0.0.0:8081
+metrics_address=http://0.0.0.0:8082
+NUM_WORKERS=1
+number_of_gpu=1
+install_py_dep_per_model=true
+number_of_netty_threads=32
+job_queue_size=1000
+model_store=/home/model-server/shared/model-store
+model_snapshot={"name":"startup.cfg","modelCount":1,"models":{"bert":{"1.0":{"defaultVersion":true,"marName":"BERTSeqClassification.mar","minWorkers":2,"maxWorkers":3,"batchSize":1,"maxBatchDelay":100,"responseTimeout":120}}}}
+```
+
+## Copy Mar file and config.properties to PVC
+
+```bash
+kubectl exec --tty pod/model-store-pod -- mkdir /pv/model-store/
+kubectl cp BERTSeqClassification.mar model-store-pod:/pv/model-store/BERTSeqClassification.mar
+
+kubectl exec --tty pod/model-store-pod -- mkdir /pv/config/
+kubectl cp config.properties model-store-pod:/pv/config/config.properties
+```
+
+## Build Torchserve image
+
+1. Clone Torchserve Repo
+
+```bash
+git clone https://github.com/pytorch/serve.git
+cd serve/docker
+```
+
+2. Modify Python and Pip paths in `Dockerfile` as below
+
+```bash
+sed -i 's#/usr/bin/python3#/opt/conda/bin/python3#g' Dockerfile
```

Comment:
Is this a must?

Reply:
Yes the ngv pytorch containers use python from conda.